NewHaven/src/Resources.cpp

```diff
@@ -0,0 +1,137 @@
+#include "Resources.h"
+#include <cstdlib>
+
+HarvestTile::HarvestTile()
+{
+	this->generateResources();
+}
+
+HarvestTile::~HarvestTile()
+{
+	delete[] this->tileContent;
+	delete[] this->position;
+}
+
+void HarvestTile::generateResources()
+{
+	std::uint_least8_t sheepCount{ 0 }, stoneCount{ 0 }, wheatCount{ 0 }, woodCount{ 0 };
+
+	for (std::uint_least8_t i = 0; i < 4; i++)
+	{
+		switch (std::rand() % 4 + 1)
```

Comment:
std::rand() is not appropriate here. The rand() function does not generate a truly random number; it actually returns the next pseudo-random value in a sequence of values ranging from 0 to RAND_MAX. This causes all 60 cards in the deck to be identical since `rand()` returns the same value throughout the lifetime of the program. You will need to import the c++ 11 random library and replace `std::rand` with `random().` If you do so this will allow your tiles to be properly randomly generated. 

However, this will cause the possibility of four different resource types appearing in a single tile. The assignment guidelines requirement
> each tile is divided into 4 squares showing 2 or 3 different Resources. 

is not met.

Reply:
I fixed the random number generator, the seed is now also pseudo-randomized for every tile generation. The tiles now look somewhat random.

If you look closely at the switch cases, there can not be more than 3 of the same resources on a tile, I have a counter for each resource, when the count reaches 3 and the random number generator generates the same thing again, the result is ignored and i is decremented for the loop to run again until the result is different.

Reply:
![Screenshot_20200214_131328](https://user-images.githubusercontent.com/26858577/74556336-d1c8b080-4f2b-11ea-933f-f6b53ba83180.png)

So I pulled the latest changes you did and ran your Driver. As seen in the above screenshot, it sometimes occur where your tile contains one of each type of resources. At least from what I understood of the instructions there should be no more than 3 different resource type on a tile.

Reply:
Oh alright, I misunderstood you and I misunderstood the instructions too. I'll make 3 different resources on a tile a maximum.

Reply:
Yeah ^^ besides that everything seems fine. Maybe add just a few comments here and there to satisfy the "documenting" portion of the grading.

Reply:
Don't worry about that, I'm trying to write up everything and then after comment the explanations to everything at the same time.